Graph metadata router info merge

buma opened the pull request:
Convex hull is generated at Graph Build time. Working on #1963.
RouterInfo now includes everything GraphMetadata previously included.
GraphMetada is now just a skeleton of itself because only envelope calculation remained in it. I renamed it to GraphEnvelope. And I'm not sure if I should leave it or move GraphEnvelope calculations to Graph itself.

I moved listing of used transitModes to Graph class itself, because I think conceptually they belong there (previously they were made in Graph Metadata).

I also updated client, analyst and profile testers with new API calls.

Current problem is that RouterInfo is newly constructed on each request (same as it was before. Only now hull and envelope are just read from Graph not made on the fly) because in RoutersTest routerId of registerGraph call and one given in GraphSource isn't the same so RouterInfo can't be created in Router constructor 

I'm not sure if this is intended or not. And if routerId is stable during OTP run. If it is I can change this to create RouterInfo in Router constructor Itself and then on getRouterInfo it is only read.

abyrd commented:
I don't think there's any deep reason why RouterInfo is constructed and filled in on demand, they were just implemented at different times for different reasons. When the RouterInfo is filled in, the values are all pulled from the Graph so it's a light-weight operation. This also avoids needing to update a cached RouterInfo whenever a graph is reloaded for a router, so I'd say it's safer to just leave it like this.

So, this is not really work in progress, we can say it's finished.

abyrd commented:
There are some deprecated functions such as getMinLatitude still present. We decided to remove those since we're already making breaking changes to the router info API.

buma commented:
I removed deprecated API calls now.

abyrd commented on src/main/java/org/opentripplanner/api/resource/GraphEnvelope.java:
```diff
@@ -0,0 +1,94 @@
+     *
+     * In {@link #updateEnvelope()} replaces envelope.
+     */
+    private transient WorldEnvelope newEnvelope;
```
We should probably call these fields osmEnvelope and osmAndTransitEnvelope for clarity.

abyrd replied:
However this is all part of an optimization that is no longer used. We should just drop it and use only a single envelope. I would even be in favor of dropping this class entirely -- it has become essentially a holder for a few (potentially static) helper functions.

abyrd commented on src/main/java/org/opentripplanner/routing/graph/Graph.java:
```diff
@@ -945,19 +962,19 @@ public void summarizeBuilderAnnotations() {
 
-    public GraphMetadata getMetadata() {
+    public GraphEnvelope getMetadata() {
         // Lazy-initialize the graph metadata since it is not serialized.
```
The comment still says "metadata" which is not wrong since the envelope is metadata, but still should probably be changed to say "envelope".

buma commented:
I removed GraphEnvelope class now. It seems that check if transit stop is in OSM envelope was used in tagged stops but I removed it anyway.

I also tested creation with OSM data, Transit data and both. And it works in all three cases. Tests also don't fail.

abyrd commented:
Thanks @buma does that mean that you consider this ready for merge now?

buma commented:
Yes. Since the check if transit stop is inside OSM envelope isn't so important in tagged stop  since code is searching couple of meters around existing stop and is not expanding.
